src/components/Input/Toggle/Toggle.styles.js

```diff
@@ -0,0 +1,125 @@
+/* stylelint-disable no-duplicate-selectors */
+import styled from "styled-components";
+import { colors, constants } from "../../..//theme";
+
+const SIZES = {
+  small: {
+    track: `
+        width: 30px;
+        height: 16px;
+        border: 1px solid;
+    `,
+    thumb: `
+        width: 10px;
+        height: 10px;
+    `,
+    activeAreaWidth: "44px",
+    offsetRange: ["3px", "17px"]
+  },
+  regular: {
+    track: `
+      width: 48px;
+      height: 24px;
+      border: 1px solid;
+    `,
+    thumb: `
+      width: 18px;
+      height: 18px;
+    `,
+    activeAreaWidth: "73px",
+    offsetRange: ["3px", "27px"]
+  }
+};
+
+const COLORS = {
+  active: {
+    border: colors.emerald.base,
+    background: colors.emerald.base,
+    thumb: colors.white.base
+  },
+  inactive: {
+    border: colors.slate,
+    background: colors.white.base,
+    thumb: colors.slate
+  },
+  activeDisabled: {
+    border: colors.diatomite,
+    background: colors.emerald.restrained,
+    thumb: colors.moonstone
+  },
+  inactiveDisabled: {
+    border: colors.diatomite,
+    background: colors.white.muted,
+    thumb: colors.moonstone
+  }
+};
+
+export const ReactToggleTrack = styled.div`
+  ${({ size }) => SIZES[size].track} padding: 0;
+  border-radius: 21px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  transition: background-color 0.3s ${constants.easing.easeInOutQuad};
+  background-color: ${({ variant }) => COLORS[variant].background};
+  border-color: ${({ variant }) => COLORS[variant].border};
+
+  &:focus {
+    border-color: ${colors.azure.base};
+    box-shadow: 0 0 5px 0 ${colors.azure.base};
+    outline: none;
+  }
+`;
+
+export const ActiveArea = styled.div`
```

Comment:
I think we should introduce this as a `button` element. This will help you leverage the `disabled` prop on button to stop event handlers from firing.

Reply:
Done